### PR TITLE
Move numeric conversions out to a new resolver

### PR DESCRIPTION
**Pull request by roji**

### CODE REVIEW

File: src/Npgsql/Internal/Resolvers/ExtraConversionsResolver.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
+using System;
+using Npgsql.Internal.Converters;
+using Npgsql.PostgresTypes;
+
+namespace Npgsql.Internal.Resolvers;
+
+sealed class ExtraConversionsResolver : IPgTypeInfoResolver
```
Comment (NinoFloris) [inline src/Npgsql/Internal/Resolvers/ExtraConversionsResolver.cs:7]: All of these probably need to be unsealed to support the inheritance pattern to opt into arrays (though I can't say for sure it'll end up working like that yet)

File: src/Npgsql/NpgsqlSlimDataSourceBuilder.cs [hunk 2]
```diff
@@ -355,7 +355,16 @@ public NpgsqlSlimDataSourceBuilder EnableRecords()
     /// </summary>
     public NpgsqlSlimDataSourceBuilder EnableFullTextSearch()
     {
-        // AddTypeInfoResolver(new FullTextSearchTypeInfoResolver());
+        AddTypeInfoResolver(new FullTextSearchTypeInfoResolver());
+        return this;
+    }
+
+    /// <summary>
+    /// Sets up mappings for extra conversions from PostgreSQL to .NET types.
```
Comment (NinoFloris) [inline src/Npgsql/NpgsqlSlimDataSourceBuilder.cs:363]: Nice and vague 😅 maybe mention numeric conversions?

File: src/Npgsql/Internal/Converters/Primitive/DoubleConverter.cs [hunk 3]
```diff
@@ -1,16 +1,43 @@
+using System.Numerics;
+
 // ReSharper disable once CheckNamespace
 namespace Npgsql.Internal.Converters;
 
-sealed class DoubleConverter : PgBufferedConverter<double>
+sealed class DoubleConverter<T> : PgBufferedConverter<T>
```
Comment (NinoFloris) [inline src/Npgsql/Internal/Converters/Primitive/DoubleConverter.cs:6]: It's back!

**Review by NinoFloris — APPROVED** [review]

File: src/Npgsql/Internal/Converters/Primitive/DoubleConverter.cs (hunk 3 above)
Comment (roji) [reply src/Npgsql/Internal/Converters/Primitive/DoubleConverter.cs:6]: Yeah, I figured if we move this out to a plugin, might as well be consistent with it :/

File: src/Npgsql/Internal/Resolvers/ExtraConversionsResolver.cs (hunk 1 above)
Comment (roji) [reply src/Npgsql/Internal/Resolvers/ExtraConversionsResolver.cs:7]: Ah, ok. Up to you - if you want I can remove them now, or you can do it when you finalize the array stuff.

File: src/Npgsql/NpgsqlSlimDataSourceBuilder.cs (hunk 2 above)
Comment (roji) [reply src/Npgsql/NpgsqlSlimDataSourceBuilder.cs:363]: We could slip in other conversions :) I don't really care, feel free to edit if you want.

**Pull request merged**